hpx/util/unlock_guard.hpp

```diff
@@ -12,6 +12,8 @@
 ///////////////////////////////////////////////////////////////////////////////
 namespace hpx { namespace util
 {
+    struct already_unlocked_t {};
+    constexpr already_unlocked_t already_unlocked = already_unlocked_t{};
     ///////////////////////////////////////////////////////////////////////////
```

Comment:
Is this type used anywhere? AFAICS, the newly added constructor `unlock_guard(Mutex& m, already_unlocked_t)` are not called from anywhere.

Reply:
Sorry. I at first misread the paper and thought we needed it. It is indeed not used anywhere.